PowerNSX.psm1

```diff
@@ -22338,7 +22376,137 @@ function Get-NsxFirewallSavedConfiguration {
 }
 
 
-########
+function Get-NsxFirewallThreshold {
+
+    <#
+    .SYNOPSIS
+    Retrieves the Distributed Firewall thresholds for CPU, Memory 
+    and Connections per Second
+
+    .DESCRIPTION
+    The firewall module generates system events when the memory and CPU usage
+    crosses these thresholds.
+
+    This command will retrieve the threshold configuration for the 
+    distributed firewall
+
+    .EXAMPLE
+
+    PS /> Get-NsxFirewallThreshold
+
+    CPU Memory ConnectionsPerSecond
+    --- ------ --------------------
+    80  80     100000
+
+
+    #>
+
+    begin {
+
+    }
+
+    process {
+        
+        
+        $URI = "/api/4.0/firewall/stats/eventthresholds"
+        $response = invoke-nsxwebrequest -method "get" -uri $URI -connection $connection
+       
+        
+        $threshold=[system.xml.xmldocument]$response.content
+
+        [PSCustomobject]@{
+                    "CPU" = $threshold.eventthresholds.cpu.percentvalue;
+                    "Memory" = $threshold.eventthresholds.memory.percentvalue;
+                    "ConnectionsPerSecond" = $threshold.eventThresholds.connectionsPerSecond.value
+                }
+    
+    }
+
+        
+       
+
+    end{}
+}
+
+function Set-NsxFirewallThreshold {
+
+    <#
+    .SYNOPSIS
+    Sets the Distributed Firewall thresholds for CPU, Memory 
+    and Connections per Second
+
+    .DESCRIPTION
+    The firewall module generates system events when the memory and CPU usage
+    crosses these thresholds.
+
+    This command will set the threshold configuration for the 
+    distributed firewall
+
+    .EXAMPLE
+
+    PS /> Set-NsxFirewallThreshold -Cpu 70 -Memory 70 -ConnectionsPerSecond 35000
+
+    CPU Memory ConnectionsPerSecond
+    --- ------ --------------------
+    70  70     35000
+
+
+    #>
+
+    param (
+
+        [Parameter (Mandatory=$true)]
+            [ValidateRange(1,100)]
+            [int]$Memory,
+        [Parameter (Mandatory=$true)]
```

Comment:
Are cpu, mem and cps mandatory?  check if API accepts single, if so - make optional.  Must test with PSBoundParameters.ContainsKey("cpu") et all before adding to xml later in cmdlet...